### ISSUE TITLE
In-place submodel editing

### ISSUE THREAD

**Issue by leozide** [leozide/leocad#222]

**Comment by Symbian9** [comment]
Very nice feature!

Just few notes:

- After enter to "Edit Selected Submodel" need change entry in context menu to something like "Edit Whole Model" or "Edit Model" (for exit from "Edit Selected Submodel" mode).
![LeoCAD Edit Submodel](http://i.imgur.com/Ci6104L.png)
- Is it possible add option for show rest of model not only in shades of gray but with transparency too?
- Is it possible add option for change rendering mode for rest of model?

**Comment by nathaneltitane** [comment]
@leozide 

So I've finally started editing a model that is complex enough to have to use the in place model editing.
The feature is quite nice although it seems to be f=very incomplete, even as of today.

After entering the in-place edit mode, there should be some way to exit it, either via the context menu, as suggested by @Symbian9 and or the addition of the proper menu entries to do so.

Also, when editing in that mode, the properties panel stays locked into the current MPD model selection and cannot let the user edit the current part being selected in-place.

So, to sum up:

1-add a clear way to exit the mode back to the upper model the edited model is nested in (context menu)
2-add all relevnt menu and shortcut entries for the in place editing
3-fix properties panel to reflect the parts being selected in the in-place model.


Thank you!

**Comment by nathaneltitane** [comment]
Hi @leozide 
Thnaks for fixing the context menu for the in-place edit mode.
Found a bug: I cannot hide selected components or parts when entering the mode.
Fix?

**Comment by Symbian9** [comment]
> Found a bug: I cannot hide selected components or parts when entering the mode.

It's look like duplication of issue https://github.com/leozide/leocad/issues/240

**Comment by Symbian9** [comment]
I just tested in-place submodel editing in `LeoCAD-git.58c6398.qt57-x86_64.AppImage` and found that there is issue with bounding box.

### Steps to reproduce

1. Create new model with one submodel

![pic.1](https://i.imgur.com/mnhAhKq.png)

2. Start in-place submodel editing (via context menu entry "Edit Selected Submodel")

![pic.2](https://i.imgur.com/wpAkk4o.png)

3. Move parts in submodel and finish in-place editing (via context menu entry "End Submodel Editing")

![pic.3](https://i.imgur.com/S6Uy28e.png)

4. RESULT: **Bounding box** of submodel and **submodel icon** (on "Parts" widget) not updated!

![pic.4](https://i.imgur.com/9jwz8iJ.png)

5. If quickly switch to submodel tab and switch back to main model tab — bounding box of submodel and submodel icon automatically fixed.

![pic.5](https://i.imgur.com/VIZCAnV.png)

@leozide, autoupdate of bounding box of submodel and submodel icon should be immediately acted after exit from in-place submodel editing mode.

**Comment by leozide** [comment]
Keeping this open, there are still bugs on this feature.
